### DISCUSSION TITLE
Managing Multiple Supabase Projects in a Next.js Application: Authentication and Data Fetching Issue

### DISCUSSION THREAD

**Discussion by wilsonibekason** [supabase/supabase#27482] [Questions]
I am using two Supabase databases/projects in my Next.js application. The first project is for an admin CMS website, which handles admin authentication and data uploads. The same Supabase project is also used to display this data on my public website. The second Supabase project is utilized for management, membership, and authentication functionalities within my application.

The issue I'm facing is that I cannot call these two Supabase projects independently in my website. Instead of using the appropriate Supabase client for each project, the application always defaults to the admin Supabase `createClient` for all requests, including authentication and data fetching for the client-side application.

![Screenshot (3)](https://github.com/supabase/supabase/assets/102219342/2d5f9080-a7da-4e06-92fc-3d7322ded9bd)

**Comment by silentworks** [comment]
You will need to set `isSingleton` to `false` otherwise it is using the same browser client for both instances.

```js
createBrowserClient<Database>(supabaseUrl, supabaseKey, { isSingleton: false });
```

Also for future write your code snippets inline instead of screenshots, it makes it easier for someone to reply with correct code sample without needing to type it all out manually.
